### ISSUE TITLE
FhirClient.Create throws NullReferenceException

### ISSUE THREAD

**Issue by wmrutten** [ewoutkramer/fhir-net-api#189]
Note: this issue was originally reported by David Hay.

Concerns FHIR API v0.90.2.33399, as currently in use by Forge for DSTU2.

When publishing a new profile to the HAPI FHIR server (http://fhirtest.uhn.ca/baseDstu2) via the method FhirClient.Create<TResource>(TResource resource), the FhirClient throw an exception:

"Exception thrown: 'System.NullReferenceException' in Hl7.Fhir.DSTU2.Core.dll"

Stack trace:

at Hl7.Fhir.Rest.FhirClient.execute[TResource](Bundle tx, IEnumerable`1 expect) in C:\Git\fhir-net-api\src\Hl7.Fhir.Core\Rest\FhirClient.cs:line 805

The exception originates from a private method in FhirClient:

```
private TResource execute<TResource>(Bundle tx, IEnumerable<HttpStatusCode> expect) where TResource : Resource
        {
            verifyServerVersion();

            var request = tx.Entry[0]; // <= Exception!!!
            var response = _requester.Execute(request);
```

It seems like the Bundle.Entry list property is empty, which is unexpected and not handled by this function.

Note that the error does not occur when publishing to the public Spark FHIR server.

**Comment by wmrutten** [comment]
Marten Smits has tested the current code base. Apparently the issue no longer occurs in the current version. I am planning to upgrade Forge to the latest API version. Afterwards, I'll re-test this bug in Forge. If it turns out to be fixed, I'll close the issue.

**Comment by brianpos** [comment]
Any updates on this one Michel?

**Comment by wmrutten** [comment]
This issue is fixed in the latest prerelease (v0.90.5-alpha1) that is now in use by Forge.
